remo/domain/dataset.py

```diff
@@ -114,7 +114,11 @@ def add_annotation(self, annotation: Annotation, image_id: int, annotation_set_i
             image_id: image id
             annotation_set_id: annotation set id
         """
-        annotation_set = self.get_annotation_set(annotation_set_id)
+        
+        if not annotation_set_id:
```

Comment:
@vmanilo fixed it as "if not annotation_set_id" in line 122.

The reason for this change is that we retrieve the whole annotation set, only if the annotation set id is not passed  - which should make the function faster?

Reply:
@drewlr in side function `get_annotation_set()` we also do check that passed annotation_set_id actually related to dataset, without this check user can try to add annotation to wrong annotation set.

but I guess it would make sense to do this check in `sdk.add_annotation()`